x/curating/spec/README.md

```diff
@@ -0,0 +1,224 @@
+# Curating Module Specification
+
+## Abstract
+
+This module provides functionality to curate social media content and earn rewards in cryptocurrency via market-based mechanisms.
+
+## Concepts
+
+First, a post is created with a `post` transaction, taking a hash of the content, and a deposit. The deposit is returned after a curation window.
+
+Posts are curated with `upvote` transactions and `moderate` transactions.
+
+Upvotes are based on quadratic voting, and take number of votes as an argument. They also takes a deposit that is returned after the curation window. Deposits are meant to keep the system honest. For example, governance may choose to slash the deposit of upvotes that are engaging in collusion or a Sybil attack.
+
+Moderate transactions are for signaling that a post contains spam or illegal content. Moderated posts immediately go to governance for adjudication. Curation is paused during this time. For posts deemed as inappropriate, curators get slashed and the moderator gets a reward. For posts not deemed as inappropriate, curation resumes while the moderator gets slashed.
+
+Only data for posts currently being curated are stored on-chain. Everything else is deleted after the curation window completes and rewards are paid out.
+
+## State
+
+### Parameters
+
+Each of these parameters can be voted on by governance, but will be assigned sane defaults for genesis.
+
+```go
+type Params struct {
+	CurationWindow       	 time.Duration
+	DefaultPostDeposit       sdk.Coin
+	DefaultUpvoteDepost  	 sdk.Coin
+	DefaultModerationDeposit sdk.Coin
+	DefaultVoteAmount        sdk.Coin // amount for 1 vote
+	MaxNumVotes              uint16 // upper-limit on QV voting
+	MaxVendors               uint16 // upper-limit on vendor_ids
+	RewardPoolAllocation 	 sdk.Dec // from inflation
+}
+```
+
+### Base Types
+
+```go
+type Post struct {
+	VendorID        uint64
+	PostID          uint64
```

Comment:
I remember @mohitmamoria mentioning something about this ID may actually be better as string but don't recall maybe he can add his comments, to me is fine as a int buy I may also see other vendors may use different type of ids (UUID?)

Reply:
Yeah was thinking about this today too. I think being a string makes covers all cases. Will change.